src/wpool_queue_manager.erl

```diff
@@ -239,8 +253,7 @@ handle_info(_Info, State) ->
 -spec get_available_worker(queue_mgr(), any(), timeout()) ->
                               noproc | timeout | {ok, timeout(), any()}.
 get_available_worker(QueueManager, Call, Timeout) ->
-    Start = now_in_milliseconds(),
-    ExpiresAt = expires(Timeout, Start),
+    ExpiresAt = expires(Timeout),
```

Comment:
I was looking at this same line for a while and wondering what to do about it 🤔 Should we stop simulating that failure maybe? I was thinking we could just be transparent about a worker not being found in time? Unless I can think of a clean way to keep the simulation working 🤔

Reply:
Ok, about this, tbh I think I might have a separate rework of the queue manager, as I think there is a race condition under which it might leak workers: if the caller crashes (an exit signal for example) just before doing `worker_ready` here https://github.com/inaka/worker_pool/blob/2f260af89b7441d383b6440f9a0383625758c1fc/src/wpool_queue_manager.erl#L81, the worker might never be given back to the queue manager and might therefore lose the worker. The time checker might have a way to rescue this after certain timeouts, but it needs configuration, and then, rescuing a worker after a timeout is a desperate way to recover them.

I think the queue manager might need to monitor clients as well as workers, but that would not only complicate code a bit but also might turn the queue manager into a bottleneck. Need to think carefully about this but all in all, I think I'd split it into a new task, wdyt?

Reply:
The failure simulation matches the overall architectural goal of this library, namely: that you can simply change all your `gen_server:…` calls to `wpool:…` ones and… magically (i.e. with the same interface and expected results and everything)… have a pool of workers where you previously had a single server. 🪄 
That's why we simulated the timeout… because _from the outside_… there was no "worker selection call"… if that fails it should look (and maybe be handled) just like if the internal `gen_server` failed.
That's also why your proposed change was looked so strange to me in the first place… I understand the purpose now but I also know that it opens the door to people doing _whatever thing_ in the runners… something that's _no at all_ an "indirect" gen_server call, as it is the intention.

Reply:
In any case, we can leave it as it is for now… and revisit the code later if we find that it is indeed problematic.